### PR TITLE
RUN-3756 - Send all IAB subscriptions to core to get success ack & emit to IAB.add(un)SubscribeListener

### PR DESCRIPTION
**Pull request by tgoc99**
[Javascript-Adapter PR](https://github.com/openfin/javascript-adapter/pull/379)

Per [RUN-3756](https://appoji.jira.com/browse/RUN-3756), was not getting any success callbacks on a 2nd IAB subscription to a given UUID (including wildcard) and topic. When fixing that, saw [this issue (RUN-3761)](https://appoji.jira.com/browse/RUN-3761) where we are also not getting the 2nd subscribe listener from IAB.addSubscribeListener so this fixes that as well since the fix for it would be a better fix than what I had for RUN-3756.  With this PR, we are letting the core handle multiple subscriptions to the same UUID & topic from a given subscriber, which allows the core to emit to any subscribe/unsubscribe listeners out there. 

[Win7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a5e61cdbd581266877d66cc)
[Win10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a5e6bf0bd581266877d66d3)

Update:
 testrunner tests: 
[subscribe - 2nd callback](https://testing-dashboard.openfin.co/#/app/tests/5a592646bd581266877d66b0/edit)
[unsubscribe - 1st callback of 2](https://testing-dashboard.openfin.co/#/app/tests/5a592648bd581266877d66b1/edit)
[addSubscribeListener - 1st/2nd subscription (get both)](https://testing-dashboard.openfin.co/#/app/tests/5a5e5c3abd581266877d66ca/edit)
[addUnsubscribeListener - 1st/2nd subscription (get both)](https://testing-dashboard.openfin.co/#/app/tests/5a5e5dcabd581266877d66cb/edit)

### CODE REVIEW

**Comment by tgoc99** [comment]
Updated, tests passed:

[Win7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a68f1ca3f32861e82b77c6d)
[Win10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a68f2733f32861e82b77c6f)

File: src/browser/api/interappbus.js [hunk 1]
```diff
@@ -134,21 +134,40 @@ function dispatchToSubscriptions(topic, identity, destUuid, destName, payload, s
...
 }
 
+function emitSubscriberAdded(identity, payload) {
+    let senderUuid = payload.sourceUuid || '*';
 
-function subscribe(identity, payload, listener) {
-    let topic = payload.topic;
-    let senderUuid = payload.sourceUuid || ANY_UUID;
-    let senderName = payload.sourceWindowName || ANY_NAME;
+    let eventingPayload = {
```
Comment (whyn07m3) [inline src/browser/api/interappbus.js]: `const`s here

File: src/browser/api/interappbus.js [hunk 2]
```diff
@@ -134,21 +134,40 @@ function dispatchToSubscriptions(topic, identity, destUuid, destName, payload, s
...
+        directMsg: payload.sourceWindowName !== '*' ? payload.sourceWindowName : false
+    };
+    busEventing.emit(ofEvents.subscriber.ADDED, eventingPayload);
+}
+
+function emitSubscriberRemoved(identity, payload) {
+    let senderUuid = payload.sourceUuid || '*';
 
-    let cbId = genCallBackId();
     let eventingPayload = {
```
Comment (whyn07m3) [inline src/browser/api/interappbus.js]: `const`s here

Comment (tgoc99) [reply src/browser/api/interappbus.js]: 👍  done

**Pull request merged**